src/vnsw/opencontrail-vrouter-netns/opencontrail_vrouter_netns/vrouter_hyperv.py

```diff
@@ -25,15 +25,98 @@ def powershell(cmds):
     return output
 
 
-class HyperVManager(object):
+LEFT_DEV_PREFIX = 'int'
+RIGHT_DEV_PREFIX = 'gw'
+
+class BlockingSSHClient(paramiko.SSHClient):
+
+    def __init__(self):
+        paramiko.SSHClient.__init__(self)
+
+    def exec_command(self, cmd):
+        _, stdout, _ = \
+            super(BlockingSSHClient, self).exec_command(cmd)
+        return stdout.channel.recv_exit_status()
+
+
+class VMPort(object):
     NAME_LEN = 14
-    WINGW_PREFIX = 'contrail-wingw-'
-    LEFT_DEV_PREFIX = 'eth'
-    RIGHT_DEV_PREFIX = 'eth'
-    PORT_TYPE = 'NameSpacePort'
+    DEV_PREFIX = 'eth'
     BASE_URL = "http://localhost:9091/port"
     HEADERS = {'content-type': 'application/json'}
 
+    def __init__(self, uuid, prefix, mac=None, ip=None):
+        self.uuid = validate_uuid(uuid)
+        if mac:
+            self.mac = netaddr.EUI(mac, dialect=netaddr.mac_eui48)
+        if ip:
+            self.ip = netaddr.IPNetwork(ip)
+
+        if prefix == LEFT_DEV_PREFIX:
+            dev_suffix = "1"
+            win_prefix = "int"
+        elif prefix == RIGHT_DEV_PREFIX:
+            dev_suffix = "2"
+            win_prefix = "gw"
+        else:
+            raise RuntimeError("prefix must be {} or {}"
+                               .format(LEFT_DEV_PREFIX, RIGHT_DEV_PREFIX))
+        self.name = self.DEV_PREFIX + dev_suffix
+        self.win_name = "{}-{}".format(win_prefix, self.uuid)[:self.NAME_LEN]
+
+    def register(self):
+        """registers wingw interface (as seen on host) to agent"""
+        # TODO do we have to register mgmt interface?
+        # get 2 snat ifaces of wingw VM
+        # _add_port_to_agent on those nics
+        pass
+
+    def unregister(self):
+        """unregisters wingw interfaces (as seen on host) from agent"""
+        # TODO do we have to unregister mgmt interface?
+        # self._delete_port_from_agent(self.nic_left)
+        # self._delete_port_from_agent(self.nic_right)
+        pass
+
+    def _request_to_agent(self, url, method, data):
```

Comment:
Stupid question - but why are you using single underscore? I think that you should use two or none.